tests/lib/rules/no-undef.js

```diff
@@ -43,6 +43,7 @@ eslintTester.addRuleTest("lib/rules/no-undef", {
         "if (typeof a === 'undefined') {}",
         { code: "var toString = 1;", ecmaFeatures: { globalReturn: true }},
         { code: "function myFunc(...foo) {  return foo;}", ecmaFeatures: { restParams: true } },
+        { code: "var App; <App />;", args: [1, {vars: "all"}], ecmaFeatures: { globalReturn: true, jsx: true } },
```

Comment:
I'm not sure this test is testing anything (same with the tests below it). Since we're not checking JSX syntax for identifiers anymore, this is effectively testing `var App;`.